docs/sync.js

```diff
@@ -35,10 +35,14 @@
  * accept or reject the server's SSL certificate.
  *
  * @property {Realm.Sync~SSLConfiguration} [ssl] - SSL configuration.
- * @property {boolean} [partial] - Whether this Realm should be opened in 'partial synchronization' mode.
- *    Partial synchronisation only synchronizes those objects that match the query specified in contrast
+ * @deprecated
+ * @property {boolean} [partial] - Whether this Realm should be opened in 'query-based synchronization' mode.
+ *    Query-based synchronisation only synchronizes those objects that match the query specified in contrast
  *    to the normal mode of operation that synchronises all objects in a remote Realm.
- *    **Partial synchronization is a tech preview. Its APIs are subject to change.**
+ * @property {boolean} [full_synchronization] - Whether this Realm should be opened in query-based or full
+ *    synchronization mode. The default is query-based mode which only synchronizes objects that have been subscribed to.
+ *    A fully synchronized Realm will synchronize the entire Realm in the background, irrespectively of the data being
+ *    used or not.
```

Comment:
Are `partial` and `full_synchronization` mutually exclusive? If so - we should prefer one over the other by warning when both are set. And perhaps we should add a `@deprecated` annotation on `partial`?

Reply:
JavaScript land prefers camelCasing - I believe this should be renamed to `fullSynchronization`.

Reply:
`partial` was marked deprecated here: https://github.com/realm/realm-js/pull/1830/files#diff-b6a6b946756f354e3a99acfa65414e2eR38

that the variable is named `full_synchronization` is a bug.